src/components/Router/Flow.js

```diff
@@ -0,0 +1,91 @@
+import { h, Component } from 'preact'
+import createHistory from 'history/createBrowserHistory'
+
+import { events } from '../../core'
+import {sendScreen} from '../../Tracker'
+import {wrapArray} from '../utils/array'
+
+const history = createHistory()
+
+class Flow extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      step: this.props.step,
+      componentsList: this.props.componentsList
+    }
+    this.unlisten = history.listen(({state}) => {
+      this.props.onStepChange(state)
```

Comment:
important to bear in mind that is the `onStepChance` props change, this is code is incomplete, since it doesn't take into account that change. Now taking into account props changing can lead no bugs down the line.

Reply:
I was wrong about this!